core/src/test/java/fj/data/EvalTest.java

```diff
@@ -39,6 +39,20 @@ public void testAlways() {
     Assert.assertEquals(tracker.getInvocationCounter(), 3);
   }
 
+  @Test
+  public void testDefer() {
+    // Make sure that a recursive computation is actually stack-safe.
+    Assert.assertEquals(even(200000).value(), "done");
+  }
+
+  private static Eval<String> even(int n) {
+    return Eval.now(n <= 0).flatMap(b -> b ? Eval.now("done") : Eval.defer(() -> odd(n - 1)));
```

Comment:
from my tests 
```
  private static Eval<String> even(int n) {
    return Eval.now(n <= 0).flatMap(b -> b ? Eval.now("done") : odd(n - 1));
  }

  private static Eval<String> odd(int n) {
    return even(n - 1);
  }
```
works equally well (since you fixed `BindTrampolineEval`) not sure when `Eval.defer` can be useful.